build_templates/buildfunctions.sh

```diff
@@ -146,24 +163,39 @@ local misc="$DART/models/utilities/ \
 
 # remove null/mpi from list
 local mpi="$DART"/assimilation_code/modules/utilities/mpi_utilities_mod.f90
+local mpif08="$DART"/assimilation_code/modules/utilities/mpif08_utilities_mod.f90
 local nullmpi="$DART"/assimilation_code/modules/utilities/null_mpi_utilities_mod.f90
 local nullwin="$DART"/assimilation_code/modules/utilities/null_win_mod.f90
 local craywin="$DART"/assimilation_code/modules/utilities/cray_win_mod.f90
 local nocraywin="$DART"/assimilation_code/modules/utilities/no_cray_win_mod.f90
+local no_cray_winf08="$DART"/assimilation_code/modules/utilities/no_cray_winf08_mod.f90
 
 if [ "$mpisrc" == "mpi" ]; then
 
    core=${core//$nullmpi/}
    core=${core//$nullwin/}
+   core=${core//$mpif08/}
```

Comment:
this looks backwards to me - is it just the labels?  mpi includes the nullmpi version, and below the \#nompi version includes the mpi modules?

Reply:
It does look backwards with the `//`  at the beginning, but if you believe me:

`core=${core//$nullmpi/}` 

replace  all matches (`//` ) of`$nullmpi` from `$core` with nothing `('/ ')`

so for mpi, remove null, nullwin, mpif08, mpif08, no_cray_winf08